metrics_v2/watcher/job_watcher.py

```diff
@@ -0,0 +1,102 @@
+import re
```

Comment:
This PR is already very large. I will move the `watcher/` code to a new PR before I finish this draft.